project-root/Assets/Scripts/Movement/Player.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+using InControl;
+
+namespace CharacterScripts
+{
+    public class Player : Character, IMoveable<CharacterController>
+    {
+        #region Variables
+        // ---------- SET THESE IN THE INSPECTOR ---------- \\
+        [Tooltip("Controls the speed at which the character is moving. Can be adjusted between a value of 0 and 20.")]
+        [Range(0, 20)] public float speed = 5f;
+
+        [Tooltip("Controls the speed at which the character is turning. Can be adjusted between a value of 0 and 20.")]
+        [Range(0, 20)] public float rotateChar = 12f;
+        // ------------------------------------------------ \\ 
+
+        // Variables that must be set at Start
+        private GameObject playerBody;
+        private GameObject playerParent;
+        private CharacterController controller;
+        private NavMeshAgent agent;
+        private Camera mainCamera;
+
+        // Player movement variables
+        private Vector3 moveDirection = Vector3.zero;
+        private Vector3 headDirection = Vector3.zero;
+
+        // Mouse turning variables
+        private int floorMask;
+        private float cameraRayLength = 100f;
+        private DeviceManager myDeviceManager;
+
+        #endregion
+
+        private void Start()
+        {
+            playerBody = gameObject.transform.GetChild(0).gameObject;

```

Comment:
I realized that this could pose a problem if the prefab hierarchy is incorrect, so this will probably be the first line that I will change.